packages/@lwc/ssr-compiler/src/compile-template/transformers/component/component.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { builders as b, is } from 'estree-toolkit';
+import { kebabcaseToCamelcase, toPropertyName } from '@lwc/template-compiler';
+import { getChildAttrsOrProps } from '../../shared';
+import { esTemplateWithYield } from '../../../estemplate';
+import { getSlottedContent } from './slotted-content';
+
+import type { BlockStatement as EsBlockStatement } from 'estree';
+import type { Component as IrComponent } from '@lwc/template-compiler';
+import type { Transformer } from '../../types';
+
+const bYieldFromChildGenerator = esTemplateWithYield`
+    {
+        const childProps = __getReadOnlyProxy(${/* child props */ is.objectExpression});
+        const childAttrs = ${/* child attrs */ is.objectExpression};
+        ${
+            /*
+                Slotted content is inserted here.
+                Note that the slotted content will be stored in a variable named `slottedContent` which
+                is used below when the child's generateMarkup function is invoked.
+            */
+            is.statement
+        }
+
+        const scopeToken = hasScopedStylesheets ? stylesheetScopeToken : undefined;
+
+        yield* ${/* generateMarkup */ is.identifier}(
+            ${/* tag name */ is.literal}, 
+            childProps, 
+            childAttrs, 
+            slottedContent,
+            instance,
+            scopeToken,
+        );
+    }
+`<EsBlockStatement>;
+
+export const Component: Transformer<IrComponent> = function Component(node, cxt) {
+    // Import the custom component's generateMarkup export.
+    const childGeneratorLocalName = `generateMarkup_${toPropertyName(node.name)}`;
+    const importPath = kebabcaseToCamelcase(node.name);
+    cxt.import({ generateMarkup: childGeneratorLocalName }, importPath);
+    cxt.import({ getReadOnlyProxy: '__getReadOnlyProxy' });
+    const childTagName = node.name;
+
+    return [
+        bYieldFromChildGenerator(
+            getChildAttrsOrProps(node.properties, cxt),
+            getChildAttrsOrProps(node.attributes, cxt),
+            getSlottedContent(node, cxt),
```

Comment:
The way you combined this together, you probably fixed this 🙁 

- https://github.com/salesforce/lwc/issues/4871

Maybe we can just let that issue hang until `engine-server` is deprecated. In the meantime I can add a test to assert the difference.

Reply:
Dang I'm wrong; it's still broken in both since both use the template compiler under the hood. Oh well 😛 https://github.com/salesforce/lwc/pull/4920